src/number_theory/cyclotomic/primitive_roots.lean

```diff
@@ -161,20 +162,44 @@ section norm
 namespace is_primitive_root
 
 variables [field L] {ζ : L} (hζ : is_primitive_root ζ n)
+variables {K} [field K] [algebra K L] [ne_zero ((n : ℕ) : K)]
+
+/-- If `finrank K L = 1`, then the norm of a `2`-th primitive root of unity is `-1`. This
+mathematically trivial result is complementary to `norm_eq_one` below. -/
+lemma norm_eq_neg_one (h : finrank K L = 1) (hζ : is_primitive_root ζ 2) :
+  norm K ζ = -1 :=
+by rw [is_primitive_root.eq_neg_one_of_two_right hζ, show -1 = algebra_map K L (-1), by simp,
+      norm_algebra_map, h, pow_one]
 
 include hζ
 
-/-- If `K` is linearly ordered (in particular for `K = ℚ`), the norm of a primitive root is `1`
-if `n` is odd. -/
-lemma norm_eq_one [linear_ordered_field K] [algebra K L] (hodd : odd (n : ℕ)) : norm K ζ = 1 :=
+/-- If `irreducible (cyclotomic n K)` (in particular for `K = ℚ`), the norm of a primitive root is
+`1` if `n ≠ 2`. -/
+lemma norm_eq_one [is_cyclotomic_extension {n} K L] (hn : n ≠ 2)
+  (hirr : irreducible (cyclotomic n K)) : norm K ζ = 1 :=
 begin
-  haveI := ne_zero.of_no_zero_smul_divisors K L n,
-  have hz := congr_arg (norm K) ((is_primitive_root.iff_def _ n).1 hζ).1,
-  rw [←(algebra_map K L).map_one , norm_algebra_map, one_pow, map_pow, ←one_pow ↑n] at hz,
-  exact strict_mono.injective hodd.strict_mono_pow hz
+  by_cases h1 : n = 1,
+  { rw [h1, one_coe, one_right_iff] at hζ,
+    rw [hζ, show 1 = algebra_map K L 1, by simp, norm_algebra_map, one_pow] },
+  { replace h1 : 2 ≤ n,
+    { by_contra' h,
+      exact h1 (pnat.eq_one_of_lt_two h) },
+    rw [← hζ.power_basis_gen K, power_basis.norm_gen_eq_coeff_zero_minpoly, hζ.power_basis_gen K,
+      ← hζ.minpoly_eq_cyclotomic_of_irreducible hirr, cyclotomic_coeff_zero _ h1, mul_one,
+      hζ.power_basis_dim K, ← hζ.minpoly_eq_cyclotomic_of_irreducible hirr, nat_degree_cyclotomic],
+    exact neg_one_pow_of_even (totient_even (lt_of_le_of_ne h1 (λ h, hn (pnat.coe_inj.1 h.symm)))) }
 end
 
-variables {K} [field K] [algebra K L] [ne_zero ((n : ℕ) : K)]
+lemma norm_of_cyclotomic_irreducible [is_cyclotomic_extension {n} K L]
+  (hirr : irreducible (cyclotomic n K)) : norm K ζ = ite (n = 2) (-1) 1 :=
+begin
+  by_cases hn : n = 2,
+  { simp only [hn, eq_self_iff_true, if_true],
+    refine norm_eq_neg_one _ (by simpa [hn] using hζ),
+    rw [is_cyclotomic_extension.finrank _ hirr, hn, pnat.coe_bit0, one_coe, totient_two],
+    apply_instance },
+  { simp only [hn, if_false, hζ.norm_eq_one hn hirr] }
```

Comment:
```suggestion
  split_ifs with hn,
  { unfreezingI {subst hn},
    convert norm_eq_neg_one hζ,
    erw [is_cyclotomic_extension.finrank _ hirr, totient_two, pow_one],
    apply_instance },
  { exact hζ.norm_eq_one hn hirr }
```

(this golf uses my change from above - if you don't like my above change, change the `convert` back to `refine`)